src/nomad_simulations/schema_packages/numerical_settings.py

```diff
@@ -890,15 +911,295 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        shape=['n_terms'],
+        description="""
+        Reference number of radial nodes for the augmented plane wave (APW) basis set.
+        This is used to derive the `energy_parameter`.
+        """,
+    )
+
+    energy_statuses = Quantity(
+        type=MEnum('fixed', 'pre-optimization', 'post-optimization'),
+        shape=['n_terms'],
+        description="""
+        Allow the code to optimize the initial energy parameter.
+        """,
+    )
+
+    differential_orders = Quantity(
+        type=np.int32,
+        shape=['n_terms'],
+        description="""
+        Derivative order of the radial wavefunction term.
+        """,
+    )  # TODO: add check non-negative # ? to remove
+
+    boundary_orders = Quantity(
+        type=np.int32,
+        shape=['n_terms'],
+        description="""
+        Differential order to which the radial wavefunction is matched at the boundary.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        if np.any(np.isneginf(self.differential_orders)):
+            logger.error('`APWLOrbital.differential_order` must be non-negative.')
+        if np.any(np.isneginf(self.boundary_orders)):
+            logger.error('`APWLOrbital.boundary_orders` must be non-negative.')
+
+
+class APWLChannel(ArchiveSection):
+    """
+    Channel of the angular momentum ($l$),
+    containing all orbitals that contribute to its description.
+    """
+
+    name = Quantity(
+        type=np.int32,
+        description="""
+        Angular momentum quantum number of the local orbital.
+        """,
+    )
+
+    n_wavefunctions = Quantity(
+        type=np.int32,
+        description="""
+        Number of wavefunctions in the l-channel, i.e. $(2l + 1) n_orbitals$.
+        """,
+    )
+
+    orbitals = SubSection(sub_section=APWWavefunction.m_def, repeats=True)
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        # self.name = self.m_def.name
+        self.n_wavefunctions = len(self.orbitals) * (2 * self.name + 1)
+
+
+class MuffinTinRegion(LocalizedBasisSet, Mesh):
+    """
+    Muffin-tin region around atoms, containing the augmented part of the APW basis set.
+    """
+
+    # there are 2 main ways of structuring the APW basis set
+    # either as APW and lo in the MT region
+    # or by l-channel in the MT region
+
+    l_max = Quantity(
+        type=np.int32,
+        description="""
+        Maximum angular momentum quantum number that is sampled.
+        Starts at 0.
+        """,
+    )
+
+    l_channels = SubSection(sub_section=APWLChannel.m_def, repeats=True)
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        self.type = 'Spherical'
+
+
+class BasisSetContainer(NumericalSettings):
+    """Section providing an overview of the full basis set.
```

Comment:
"""
A base section used to define the basis set used during the diagonalization of the Hamiltonian defined as `ModelMethod`. This section may contain multiple basis set specifications under the `basis_sets` (rename here to `basis_set_components` or something similar), each with their own parameters.
"""

Reply:
Added with only slight modifications

Reply:
Btw, I would reserve renaming `BasisSetContainer` and `BasisSet` to `FullBasisSet`/`BasisSet` and `BasisSetComponent` until we migrate the parsers.
Else the naming may become confusing.